e2e/cypress/support/api_commands.js

```diff
@@ -725,6 +725,23 @@ Cypress.Commands.add('apiUnpinPosts', (postId) => {
 // https://api.mattermost.com/#tag/system
 // *****************************************************************************
 
+Cypress.Commands.add('apiUpdateConfigBasic', (newSettings = {}) => {
```

Comment:
nit: without context, `Basic` isn't entirely clear to me. When should I use this vs. the `non-Basic` one? Also, if I happen to be testing as a non-sysadmin, this will just fail with a 403. Would it make sense to route this through a separate set of authentication tokens that's guaranteed to be the system administrator but won't affect the main login?

Reply:
re: Basic - I'm out of names.  Open for suggestion. 

The other one is a convenience function which were used when changing config. Note that most of the tests involves regular users only (so logging in then out will be too much).

Other than the use case in this test, I can see it to be used in verifying whether a logged in user/bot/token/etc has a permission to do so.

Reply:
> if I happen to be testing as a non-sysadmin, this will just fail with a 403
I should say, it's expected and so the other one should be commonly used.

Reply:
I won't block the review on this, but I think this pair of functions still leaves a testing gap: tests that assert non-sysadmin semantics in the presence of configuration changes. With the current function, I'm forced to write my test as a `sysadmin` (using `Basic` to avoid the logout as in the case here), or to fix my configuration before I login as a regular user. There's still no way to test what happens to a regular user when the configuration is changed underneath them, without also forcing them to logout.

>> if I happen to be testing as a non-sysadmin, this will just fail with a 403
> I should say, it's expected and so the other one should be commonly used.

I see these as helper functions that "sideload" effects into the system but aren't, themselves, assertions beyond their ability to fail). It should be possible for the test to change the configuration underneath of the active user, ideally by managing a completely different cookie jar / session to achieve same.

Reply:
> tests that assert non-sysadmin semantics in the presence of configuration changes

@lieut-data `cy.externalRequest` might help solve that. You may initiate config change via external request while observing the effect on current session.

Reply:
Got it -- I think I had expected the additional methods here to use `cy.externalRequest` themselves.